README.md

```diff
@@ -501,54 +543,82 @@ To create a dynamic list of counter we have to use the `list` function.
 
 ### Understanding generator functions
 
-At first Turbine's use of generator functions may seem peculiar. But
-it is just a nicer way to call [`chain`](#componentchain) methods.
+Turbine's use of generator functions may seem a bit puzzling at first.
+For instance, it may seem like generator functions serve two different
+purposes. One when they're used in the model and another when they're
+used in the view
 
-Three types in Turbine have a `chain`-method: `Component`, `Now` and
-`Behavior`. `chain` is used to compose structures together such that the result
-from one structure can be used in the next.
+But, what they do under the hood is exactly the same in both cases.
+The key to understand is that generator functions is just sugar for
+calling [`chain`](#componentchain) several times in succession.
 
-For instance, when we use `chain` on components we can combine
-elements as we'd like and pipe output from one component into the
-next. The code below combines two `input` elements with a `span`
-element that shows the concatenation of the text in the two input
-fields.
+When we use `chain` on components we can combine elements and pipe
+output from one component into the next. The code below combines two
+`input` elements with a `span` element that shows the concatenation of
+the text in the two input fields.
 
 ```typescript
 input({ attrs: { placeholder: "foo" } }).chain(
-  ({ inputValue: a }) => input().chain(
-    ({ inputValue: b }) => span(["Combined text: ", a, b])
+  ({ inputValue: aValue }) => input().chain(
+    ({ inputValue: bValue }) => {
+      const concatenated = lift((a, b) => a + b, aValue, bValue);
+      return span(["Concatenated text: ", concatenated]).mapTo({concatenated});
+    }
```

Comment:
```js
span(["Concatenated text: ", aValue, bValue])
```

Where you previously wrote the same result 
as simply passing behaviors as arguments if I understood it correctly.